caffe2/core/typeid.h

```diff
@@ -162,6 +162,8 @@ class TypeMeta {
   // in .cpp to manage dependencies
   static void _ThrowRuntimeTypeLogicError(const std::string& msg);
 
+  static CaffeTypeId _createTypeId();
```

Comment:
Should this be probably a static member function of CaffeTypeId, instead of TypeMeta?

Reply:
Good idea, this allowed me to remove the constructors from CaffeTypeId, leaving this the only way to create instances.